docs(component): pull options out of jsdoc

rwaskiewicz opened the pull request:
this commit attempts to pull the content out of the existing
`@Component()` documentation (currently in a JSDoc block) to improve the
searchability of options, and begin to unify how we display said
options with pages such as the dist-custom-elements output target.

additional examples have been added to clarify usage of the options in
`@Component()`

**All changes are currently only in the 'next' version**

**Current Asks**
- ✔️ My highest priority here is format - does this look/feel OK?
-  ✔️ Second to that is content - does that seem right?
- Take a look at v2 docs, I propagated the changes there (and to v3.0, but they're the same as 'next'). I added a small section for `assetsDir`, which we removed in v3.0.0

✔️ Once we have those two down, I'll propagate this to the other pages

**TODO**
- ✔️ Propagate this to other pages (v2, v3)

tanner-reits commented:
This looks/feels a lot better to me. Glad we're starting to standardize this across our docs. My only suggestion would be to possibly alphabetize the options... that might just be my OCD talking though 😅

rwaskiewicz commented:
@tanner-reits fair with regard to alphabetization. I kept `tag` first since it's a required field, and the rest are optional (as a doc reader, I suppose I like knowing what I _have_ to configure before I look at optional ones (and don't wanna wade through optionals to find required ones))

alicewriteswrongs reviewed (approved):
lgtm! much nicer looking 👍

rwaskiewicz commented:
@tanner-reits && @alicewriteswrongs can you give this another look?

I've added 2 commits:
- https://github.com/ionic-team/stencil-site/pull/1018/commits/fd774ea5ebb93ca5e36619f22442a8d3702ae4f1 propagates the changes to v3.0 & v2 docs (so we should have parity between all three)
- https://github.com/ionic-team/stencil-site/pull/1018/commits/ca2dd706b0df6d162962b477a390a5d39484accf adds back a small section for the v2 docs regarding `assetsDir`, which we removed in v3.0.0 of Stencil

tanner-reits reviewed (approved):
lgtm!

alicewriteswrongs reviewed (approved):
lgtm!
